subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/bytebuffer/BBType.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.persistence.typeHandling.bytebuffer;
+
+/**
+ * Type codes for array types.
+ */
+public enum BBType {
```

Comment:
this is what I think you can do. the type can be encoded into a short so 2 bytes. reserve the first byte as types and the rest can be reserved for modifiers. so you can support up to 255 different types and the modifiers can be things such as an array modifier. or some other special modifier based on the type.